font/src/lib.rs

```diff
@@ -267,3 +321,27 @@ pub trait Rasterize {
     /// Update the Rasterizer's DPI factor
     fn update_dpr(&mut self, device_pixel_ratio: f32);
 }
+
+/// Config option specific to the Rasterizer.
+/// Since the Rasterizer lives in the subcrate font we do not want it to depend on the Font config
+/// struct from alacritty, as this would introduce a circular dependency between the crates.Clone
+/// This struct specifies the subset of Font that the Rasterizer cares about, then traits can be
+/// used to convert from Font to this struct when constructing a Rasterizer.
+#[cfg(feature = "hb-ft")]
+pub struct RasterizeConfig {
+    /// Toggle thin strokes on mac osx
+    // Technically this is impossible while under "hb-ft" but is included for compatiblity in the
+    // api
+    pub use_thin_strokes: bool,
+    /// Toggle rendering of font ligatures
+    pub use_font_ligatures: bool,
+}
+
+// Only implemented for the FreeType rasterizer so far.
+/// Conceptually this extends the Rasterizer trait with Harfbuzz specific functionality.
+#[cfg(feature = "hb-ft")]
+pub trait HbFtExt {
+    /// Shape the provided text into a set of glyphs.
+    /// TODO: properly report HarfBuzz errors
```

Comment:
This is something we should probably do before merging.

We need to be clear about which errors we want to ignore/handle ourselves, which we want to just crash on and which we want to propagate to Alacritty.